dace/sdfg/state.py

```diff
@@ -1526,6 +1526,9 @@ def add_memlet_path(self,
                     if propagate:
                         cur_memlet = propagate_memlet(self, cur_memlet, snode,
                                                       True)
+        # Try to initialize memlets
+        for edge in edges:
+            edge.data.try_initialize(self.parent, self, edge)
```

Comment:
Is this related or did it sneak in?

Reply:
It's related. The optimizer CLI, even when you don't optimize, actually serialized and deserialized, fixing the memlets. Now we just fix them during creation